### DISCUSSION TITLE
libwebp CVE

### DISCUSSION THREAD

**Discussion by jmartin45-chwy** [processone/ejabberd#4096] [Questions] [answered]
Hello, we are using 23.01 which includes a version of libwebp that has a Critical CVE: https://nvd.nist.gov/vuln/detail/CVE-2023-4863

`~ $ apk version libwebp
WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.16/main: No such file or directory
WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.16/community: No such file or directory
Installed:                                Available:
libwebp-1.2.3-r0                        ?  `

Will you be releasing a new Docker image to address this?

If not, can you suggest a workaround?  I'm not a docker expert.

Thanks

**Comment by badlop** [comment, marked as answer]
The script that builds the installers (and the `ejabberd` container image) was updated to use libwebp 1.3.2 two weeks ago in commit b16530b.

The next ejabberd release (that is planned for before end of this month, hopefully next week) will include it (all the installers, the `ejabberd` container image, and the `ecs` container image).

In the mean time, it is possible to find that recent code in:
- installers: click on the latest commit https://github.com/processone/ejabberd/actions/workflows/installers.yml download the `ejabberd-packages.zip`
- If using the `ejabberd` container image, use the [`master` tag](https://github.com/processone/ejabberd/pkgs/container/ejabberd/135948459?tag=master) 
- If using the `ecs` container image: that one is manually built only for each ejabberd release

**Reply by jmartin45-chwy** [reply]
We use the ecs image unfortunately.

We don't have the thumbnail module enabled.  Are there other modules that use the webp library that we could disable in the meantime?

Thank you

**Reply by weiss** [reply]
[`mod_avatar`][1] uses libwebp code if (and only if) the `convert` option is specified to convert avatars from/to WebP format.

With default settings, ejabberd shouldn't be affected by the vulnerability, unless I'm overlooking something.

[1]: https://docs.ejabberd.im/admin/configuration/modules/#mod-avatar

**Reply by jmartin45-chwy** [reply]
Perfect, thank you
